cli/azd/pkg/project/internal/project_utils.go

```diff
@@ -34,3 +37,25 @@ func CreateDeployableZip(appName string, path string) (string, error) {
 
 	return zipFile.Name(), nil
 }
+
+const c_gitIgnore string = ".gitignore"
+
+// CreateSkipPatternsFromGitIgnore inspect root project path and a `servicePath`
+// to see if there is a .gitignore file. It then combine both files in a single list
+// of exclusions.
+func CreateSkipPatternsFromGitIgnore(servicePath string) ([]gitignore.GitIgnore, error) {
+	// azdContext will provide the azd-project root path
+	azdContext, err := azdcontext.NewAzdContext()
+	if err != nil {
+		return nil, err
+	}
+
+	rootPath := azdContext.ProjectDirectory()
+	// If .gitignore can't be open is fine, it could be missing
+	var allPatterns []gitignore.GitIgnore
+	rootPatterns, _ := gitignore.NewFromFile(filepath.Join(rootPath, c_gitIgnore))
+	servicePatterns, _ := gitignore.NewFromFile(filepath.Join(servicePath, c_gitIgnore))
```

Comment:
A `.gitignore` can exist any and folder in the structure and would be applied.  Looks like we are only looking at the project root dir. 

Also, many framework languages have their own concepts for this.

NPM - https://www.npmjs.com/package/npmignore
Docker - https://docs.docker.com/engine/reference/builder/#dockerignore-file

This feels to me that it should be a per `FrameworkService` implementation and have our own defaults if nothing is specified.  `.gitignore` doesn't feel like the appropriate option here.

Reply:
This change only applies to `building python and npn` , as those are the only ones which build process is only `copying src files to a new folder`.   

However, this approach can scale to any framework service.

It also scales to introduce an `.azddeployignore` file if we want, in case we want, for example, to exclude folders like `\test` for creating a build output from the `copy - paste` strategy.